include/flatbuffers/flatbuffers.h

```diff
@@ -2142,13 +2142,30 @@ struct TypeTable;
 // Signature of the static method present in each type.
 typedef TypeTable *(*TypeFunction)();
 
+// Key-value structure as used by an attribute
+struct AttributeKeyValue {
+  const char *key;
+  const char *value;
+};
+
+// Key-value pair array for storing custom attribute values.
+template<size_t count> struct AttributeList {
+  size_t num_elems;  // of the two arrays below.
+  AttributeKeyValue attributes[count];
```

Comment:
my idea was that this field would be a pointer too, but I guess an array of size 1 can work too.